iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/transformation/dag/input/ConstantInputReader.java

```diff
@@ -21,55 +21,73 @@
 
 import org.apache.iotdb.db.exception.query.QueryProcessException;
 import org.apache.iotdb.db.queryengine.plan.expression.leaf.ConstantOperand;
-import org.apache.iotdb.db.queryengine.transformation.api.LayerPointReader;
+import org.apache.iotdb.db.queryengine.transformation.api.LayerReader;
 import org.apache.iotdb.db.queryengine.transformation.api.YieldableState;
 import org.apache.iotdb.db.utils.CommonUtils;
 
 import org.apache.commons.lang3.Validate;
+import org.apache.tsfile.block.column.Column;
+import org.apache.tsfile.common.conf.TSFileDescriptor;
 import org.apache.tsfile.enums.TSDataType;
+import org.apache.tsfile.read.common.block.column.BinaryColumn;
+import org.apache.tsfile.read.common.block.column.BooleanColumn;
+import org.apache.tsfile.read.common.block.column.DoubleColumn;
+import org.apache.tsfile.read.common.block.column.FloatColumn;
+import org.apache.tsfile.read.common.block.column.IntColumn;
+import org.apache.tsfile.read.common.block.column.LongColumn;
+import org.apache.tsfile.read.common.block.column.RunLengthEncodedColumn;
 import org.apache.tsfile.utils.Binary;
 
 import java.io.IOException;
+import java.util.Optional;
 
-/** LayerPointReader for constants. */
-public class ConstantInputReader implements LayerPointReader {
+public class ConstantInputReader implements LayerReader {
+  private final TSDataType dataType;
 
-  private final ConstantOperand expression;
-
-  protected int cachedInt;
-  protected long cachedLong;
-  protected float cachedFloat;
-  protected double cachedDouble;
-  protected boolean cachedBoolean;
-  protected Binary cachedBinary;
+  private final Column cachedColumn;
 
   public ConstantInputReader(ConstantOperand expression) throws QueryProcessException {
-    this.expression = Validate.notNull(expression);
+    Validate.notNull(expression);
```

Comment:
This is a Deprecated method. You could add some error msg.

Reply:
Done.